js/src/scrollspy.js

```diff
@@ -133,7 +133,7 @@ const ScrollSpy = (($) => {
             target = $(targetSelector)[0]
           }
 
-          if (target && (target.offsetWidth || target.offsetHeight)) {
+          if (target && (target.getBoundingClientRect().width || target.getBoundingClientRect().height)) {
```

Comment:
I don't know how works the logic of ``getBoundingClientRect`` but you can declare variable to just have one call to ``getBoundingClientRect``